Libraries/Connect/Internal/Utilities/TimeoutTimer.swift

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022-2024 The Connect Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+final class TimeoutTimer: @unchecked Sendable {
+    private var hasTimedOut = false
+    private var onTimeout: (() -> Void)?
+    private let queue = DispatchQueue(label: "rpc.connect.Timeout")
```

Comment:
nit: the name looks like it's a timeout for establishing a connection. Maybe `connectrpc.Timeout`?

Reply:
updated